plugins/by-name/clangd-extensions/default.nix

```
@@ -1,271 +1,161 @@
-{
-  lib,
-  helpers,
-  config,
-  pkgs,
-  ...
-}:
-with lib;
+{ config, lib, ... }:
 let
-  cfg = config.plugins.clangd-extensions;
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types mapAttrs;
+in
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "clangd-extensions";
+  moduleName = "clangd_extensions";
+  packPathName = "clangd_extensions.nvim";
+  package = "clangd_extensions-nvim";
+  description = ''
+    Clangd's off-spec features for neovim's LSP client.
+  '';
 
-  basePluginPath = [
-    "plugins"
-    "clangd-extensions"
-  ];
+  maintainers = [ lib.maintainers.jolars ];
 
-  borderOpt = helpers.defaultNullOpts.mkBorder "none" "clangd-extensions" "";
-in
-{
   # All of those warnings were introduced on 08/22/2023.
   # TODO: Remove them in ~2 months (Oct. 2023).
-  imports =
-    [
-      (mkRemovedOptionModule (basePluginPath ++ [ "server" ]) ''
-        To configure the `clangd` language server options, please use
-        `plugins.lsp.servers.clangd.extraSettings`.
-      '')
-      (mkRemovedOptionModule (
-        basePluginPath
-        ++ [
-          "extensions"
-          "autoSetHints"
-        ]
-      ) "")
-    ]
-    ++ (map
-      (
-        optionPath:
-        mkRenamedOptionModule (basePluginPath ++ [ "extensions" ] ++ optionPath) (
-          basePluginPath ++ optionPath
-        )
-      )
-      [
-        [
-          "inlayHints"
-          "inline"
-        ]
-        [
-          "inlayHints"
-          "onlyCurrentLine"
-        ]
-        [
-          "inlayHints"
-          "onlyCurrentLineAutocmd"
-        ]
-        [
-          "inlayHints"
-          "showParameterHints"
-        ]
-        [
-          "inlayHints"
-          "parameterHintsPrefix"
-        ]
-        [
-          "inlayHints"
-          "otherHintsPrefix"
-        ]
-        [
-          "inlayHints"
-          "maxLenAlign"
-        ]
-        [
-          "inlayHints"
-          "maxLenAlignPadding"
-        ]
-        [
-          "inlayHints"
-          "rightAlign"
-        ]
-        [
-          "inlayHints"
-          "rightAlignPadding"
-        ]
-        [
-          "inlayHints"
-          "highlight"
-        ]
-        [
-          "inlayHints"
-          "priority"
-        ]
-        [ "ast" ]
-        [ "memoryUsage" ]
-        [ "symbolInfo" ]
-      ]
-    );
+  imports = [ ./deprecations.nix ];
 
-  options.plugins.clangd-extensions = lib.nixvim.plugins.neovim.extraOptionsOptions // {
-    enable = mkEnableOption "clangd_extensions, plugins implementing clangd LSP extensions";
+  # TODO: introduced 2025-01-08: remove by 25.05
```

```suggestion
  # TODO: introduced 2025-01-08: remove after 25.05
```

Subtle wording nit; it's not that we must remove _before_ (by), but that we must not remove until after 25.05 at the earliest.

Or 24.05 at the earliest for the pre-existing deprecations.